myhoard/restore_coordinator.py

```diff
@@ -450,6 +450,15 @@ def rebuild_tables(self) -> None:
         self._ensure_mysql_server_is_started(with_binlog=False, with_gtids=False)
         # Rebuilding very large tables can be slow
         with self._mysql_cursor(timeout=3600.0) as cursor:
+            cursor.execute("SELECT @@SESSION.sql_mode")
+            unwanted_sql_modes = {"STRICT_ALL_TABLES", " STRICT_TRANS_TABLES", "NO_ZERO_DATE", "NO_ZERO_IN_DATE"}
+            sql_mode = cursor.fetchone()["@@SESSION.sql_mode"]
+            lenient_sql_mode = ",".join([mode for mode in sql_mode.split(",") if mode not in unwanted_sql_modes])
+            if lenient_sql_mode != sql_mode:
```

Comment:
Don't we need some sort of sorting to ensure we're not tripped-up on different sort-orders?

Reply:
the `unwanted_sql_modes` is an unsorted set, but the split and comprehension list to edit the `sql_mode` itself keep the existing order, so I don't expect any issue there.